testsuite/testcases/src/state_sync_performance.rs

```diff
@@ -22,73 +28,93 @@ impl Test for StateSyncFullnodePerformance {
 
 impl NetworkTest for StateSyncFullnodePerformance {
     fn run<'t>(&self, ctx: &mut NetworkContext<'t>) -> Result<()> {
-        // Verify we have at least 1 fullnode
-        let all_fullnodes = ctx
-            .swarm()
-            .full_nodes()
-            .map(|v| v.peer_id())
-            .collect::<Vec<_>>();
-        if all_fullnodes.is_empty() {
-            return Err(anyhow::format_err!(
-                "Fullnode performance tests require at least 1 fullnode!"
-            ));
-        }
+        let all_fullnodes = get_fullnodes_and_check_setup(ctx, self.name())?;
```

Comment:
changes here https://github.com/aptos-labs/aptos-core/pull/5490/files#diff-a83aaea40a44dc1164c9cdc3bf5882913d55a755ba1fa82f15643a826e35dd50 might simplify this logic?

Reply:
Nice, I'll clean it up when I do my next forge related task 😄